core/src/main/java/org/frankframework/validation/SchemaUtils.java

```diff
@@ -223,217 +240,260 @@ private static void addXsdMergeWarnings(final IScopeProvider scopeProvider, fina
+		return collectedAttributes
+				.stream()
+				.noneMatch(attribute2 -> XmlUtils.attributesEqual(attribute, attribute2));
+	}
+
 	/**
+	 * Internal method used for XSD merging and rewriting. The method has two modes: one to collect information from
+	 * all XSDs that need to be merged, and one to produce parts of the merged XSD.
+	 * <p>
 	 * Including a {@link IXSD} into an {@link XMLStreamWriter} while parsing it. It is parsed
 	 * (using a low level {@link XMLEventReader}) so that certain things can be corrected on the fly.
+	 * </p>
 	 *
-	 * @param xsd
-	 * @param xmlStreamWriter
-	 * @param standalone When standalone the start and end document contants are ignored, hence the xml declaration is ignored.
-	 * @param stripSchemaLocationFromImport Useful when generating a WSDL which should contain all XSD's inline (without includes or imports).
-	 *        The XSD might have an import with schemaLocation to make it valid on it's own, when stripSchemaLocationFromImport is true it will be removed.
+	 * @param xsd                     The XSD to write to the {@link XMLStreamWriter}.
+	 * @param xmlStreamWriter         The {@link XMLStreamWriter} to use for output. If {@code null}, then only parse the XSD collect information about imports and attributes.
+	 * @param skipRootStartElement    When writing merged XSDs, only when writing the first XSD the root start element should be included.
+	 * @param skipRootEndElement      When writing merged XSDs, only when writing the last XSD the root start element should be included.
+	 * @param rootAttributes          List to which to collect information on attributes on the XSD root element, or from which to write the XSD root element attributes, depending on the mode.
+	 * @param rootNamespaceAttributes List to which to collect root namespace attributes, or from which to write the root namespace attributes, depending on the mode.
+	 * @param imports                 List to which to collect XSD imports, or from which to write all XSD imports, depending on the mode.
 	 */
-	public static void xsdToXmlStreamWriter(final IXSD xsd, XMLStreamWriter xmlStreamWriter, boolean standalone, boolean stripSchemaLocationFromImport, boolean skipRootStartElement, boolean skipRootEndElement, List<Attribute> rootAttributes, List<Namespace> rootNamespaceAttributes, List<XMLEvent> imports, boolean noOutput) throws IOException, ConfigurationException {
-		Map<String, String> namespacesToCorrect = new HashMap<>();
-		NamespaceCorrectingXMLStreamWriter namespaceCorrectingXMLStreamWriter = new NamespaceCorrectingXMLStreamWriter(xmlStreamWriter, namespacesToCorrect);
+	private static void xsdToXmlStreamWriter(final @Nonnull IXSD xsd, @Nonnull XMLStreamWriter xmlStreamWriter, boolean skipRootStartElement, boolean skipRootEndElement, @Nonnull List<Attribute> rootAttributes, @Nonnull List<Namespace> rootNamespaceAttributes, @Nonnull List<XMLEvent> imports) throws IOException, ConfigurationException {
```

Comment:
This method has been simplified to only write to the XSD, not parse and collect information. Many options removed and duplicate code extracted.